src/api.rs

```diff
@@ -223,7 +223,21 @@ impl Context {
         return false;
       }
 
-      let slot_idx = self.fi.order_hint % REF_FRAMES as u32;
+      let lvl = if !reorder {
+        0
+      } else if idx_in_group < pyramid_depth {
+        idx_in_group
+      } else {
+        pyramid_depth - (idx_in_group - pyramid_depth + 1).trailing_zeros() as u64
```

Comment:
Maybe you want to use [saturating_sub](https://doc.rust-lang.org/std/primitive.u64.html#method.saturating_sub) as you did below? 

Probably a comment on what is the logic behind would be nice :)